### PR TITLE
Upstream changes

### PR DESCRIPTION
**Pull request by krnlyng**

### CODE REVIEW

**Review by sledges — APPROVED** [review]
:+1: BTW there's also a follow-up in the workings (upstream PR#1825):)

**Comment by krnlyng** [comment]
@sledges ah yes, we can wait until that one is merged. should appear automatically :).

**Review by mkosola — APPROVED** [review]

**Pull request merged**
